Packs/MobileIronUEM/Integrations/MobileIronCLOUD/MobileIronCLOUD.py

```diff
@@ -0,0 +1,511 @@
+
+    return incidents
+
+
+def execute_fetch_incidents_command(client):
+    last_run = demisto.getLastRun()
+    datetime_now = datetime.utcnow()
+    datetime_now_iso = datetime_now.isoformat()
+
+    params = demisto.params()
+    fetch_interval = int(params.get('fetch_interval'))
+    max_fetch = int(params.get('max_fetch'))
+
+    should_run = should_run_fetch_incidents(last_run, datetime_now, fetch_interval)
+
+    if should_run:
+        partition_id = get_partition_id(client)
+        incident_type = params.get('incidentType')
+        incidents = fetch_incidents(client=client, partition_id=partition_id, incident_type=incident_type,
+                                    max_fetch=max_fetch)
+        demisto.incidents(incidents)
+        demisto.setLastRun({'time': datetime_now_iso})
+    else:
+        demisto.incidents([])
+
+
+'''MAIN FUNCTION'''
+
+
+def main():
+    params = demisto.params()
+    args = demisto.args()
+    credentials = params.get('credentials')
+    username = credentials.get('identifier')
+    password = credentials.get('password')
+    base_url = params.get('url')
+
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    try:
+        client = MobileIronCloudClient(
+            base_url=base_url,
+            verify=verify_certificate,
+            auth=(username, password),
+            proxy=proxy)
+        command_methods = {
+            'test-module': lambda: execute_test_module_command(client),
+            'fetch-incidents': lambda: execute_fetch_incidents_command(client),
+            'mobileiron-cloud-get-devices-data': lambda: execute_get_devices_data_command(client),
+            'mobileiron-cloud-get-device-by-mac': lambda:
+            execute_get_device_by_field_command(client, field_name='wifiMacAddress',
+                                                field_value=args.get("device_mac")),
+            'mobileiron-cloud-get-device-by-serial': lambda:
+            execute_get_device_by_field_command(client, field_name='serialNumber',
+                                                field_value=args.get("device_serial")),
+            'mobileiron-cloud-get-device-by-id': lambda: execute_get_device_by_id_command(client),
+            'mobileiron-cloud-unlock-device': lambda: execute_device_action_command(client, "unlock"),
+            'mobileiron-cloud-retire-device': lambda: execute_device_action_command(client, "retire"),
+            'mobileiron-cloud-wipe-device': lambda: execute_device_action_command(client, "wipe"),
+            'mobileiron-cloud-force-check-in': lambda: execute_device_action_command(client, "forceCheckin"),
+            'mobileiron-cloud-send-message': lambda: execute_send_message_command(client)
+        }
+        command_method = command_methods.get(demisto.command())
+        if not command_method:
+            raise DemistoException(f'command not recognised - {demisto.command()}')
+
+        result = command_method()
```

Comment:
to me personally this reads much better and its more concise than an if-else chain especially since pretty much all commands have a return_result statement, but i will change it back. It probably makes sense as all other integrations i saw have an if-else chain for handling commands